cmake/util.cmake

```diff
@@ -21,19 +20,14 @@ function(target_circle_link_libraries TARGET_NAME)
                     list(APPEND LIBS "-Wl,-force_load")
                 endif()
                 list(APPEND LIBS ${arg})
-                list(APPEND libsInArgn ${arg})
             endif()
         endforeach()
         if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang")
             if(NOT IOS_ENABLE_BITCODE)
                 list(APPEND LIBS "-undefined dynamic_lookup")
             endif()
         endif()
-        list(REVERSE libsInArgn)
```

Comment:
Just for curiosity, why don't we need `list(REVERSE libsInArgn)` anymore?

Reply:
As shown in #4892, on Mac platform I found so many libraries are linked twice, one in a given order, another in the reversed order. 
```
-Wl,-force_load ../../gserver/libpaddle_gserver.a -Wl,-force_load ../../function/libpaddle_function.a ../../pserver/libpaddle_pserver.a ../../trainer/libpaddle_trainer_lib.a ../../pserver/libpaddle_network.a ../../math/libpaddle_math.a ../libpaddle_utils.a ../../parameter/libpaddle_parameter.a ../../../proto/libpaddle_proto.a ../../cuda/libpaddle_cuda.a ../../optimizer/libpaddle_optimizer.a 
../../../../third_party/install/gflags/lib/libgflags.a
../../../../third_party/install/glog/lib/libglog.a
/usr/local/opt/openblas/lib/libopenblas.dylib
../../../../third_party/install/protobuf/lib/libprotobuf.a
../../../../third_party/install/zlib/lib/libz.a
/usr/lib/libpython2.7.dylib
-undefined dynamic_lookup
/usr/lib/libpython2.7.dylib
../../../../third_party/install/zlib/lib/libz.a
../../../../third_party/install/protobuf/lib/libprotobuf.a
/usr/local/opt/openblas/lib/libopenblas.dylib
../../../../third_party/install/glog/lib/libglog.a
../../../../third_party/install/gflags/lib/libgflags.a
../../optimizer/libpaddle_optimizer.a ../../cuda/libpaddle_cuda.a ../../../proto/libpaddle_proto.a ../../parameter/libpaddle_parameter.a ../libpaddle_utils.a ../../math/libpaddle_math.a ../../pserver/libpaddle_network.a ../../trainer/libpaddle_trainer_lib.a ../../pserver/libpaddle_pserver.a ../../function/libpaddle_function.a ../../gserver/libpaddle_gserver.a
```

Here, `LIBS` contains all dependent libraries in given order, `libsInArgn` contains all the dependent in the reverse order. The reverse is done by `line 32`. And both `LIBS` and `libsInArgn` are linked to `TARGET_NAME`.  I don't know why it was written like this. When I remove `libsInArgn`, this PR also works on Mac.